clients/stellar-relay-lib/src/tests/mod.rs

```diff
@@ -43,31 +40,45 @@ async fn stellar_overlay_should_receive_scp_messages() {
 	let (node_info, conn_info) = overlay_infos(false);
 
 	let overlay_connection = Arc::new(Mutex::new(
-		StellarOverlayConnection::connect(node_info, conn_info).await.unwrap(),
+		StellarOverlayConnection::connect(node_info, conn_info)
+			.await
+			.expect("should connect"),
 	));
+
+	// to check if we receive any scp message from stellar node
+	let (sender, receiver) = tokio::sync::oneshot::channel();
 	let ov_conn = overlay_connection.clone();
 
 	let scps_vec = Arc::new(Mutex::new(vec![]));
 	let scps_vec_clone = scps_vec.clone();
-
-	timeout(Duration::from_secs(300), async move {
+	tokio::spawn(async move {
 		let mut ov_conn_locked = ov_conn.lock().await;
-		if let Ok(Some(msg)) = ov_conn_locked.listen() {
-			scps_vec_clone.lock().await.push(msg);
-
-			ov_conn_locked.stop();
+		loop {
```

Comment:
We have to loop here because `fn listen()` does not have a loop inside it, anymore.

Reply:
We could also `.await().expect()` the spawn handle, maybe then we don't need the other `timeout` thread. I read that maybe a panic inside a thread may not make the test fail.

Reply:
Updated it. :+1: